Initial commit

dubeyShivank opened the pull request.

cTopher commented on src/pico.css:
I would completely remove this file

cTopher commented on src/lib/model.ts:
```diff
@@ -0,0 +1,19 @@
+	readonly jwtUrl?: string
+}
+
+export async function getConfig(code: string): Promise<DemoConfiguration> {
```
This entire file is not used and can be removed.

cTopher commented on package.json:
```diff
@@ -0,0 +1,41 @@
+	},
+	"type": "module",
+	"dependencies": {
+		"@picocss/pico": "^1.5.10",
```
Other styling remark: I would copy the overlay effect used in our zapper.

![image](https://github.com/user-attachments/assets/c8f40c0d-808a-4665-9674-1360410c0db5)

A table could be as simple as:
```
table {
		padding: 1rem;
		margin: 1rem;
		background: rgba(0, 0, 0, 0.5);
	}

	th {
		text-align: left;
	}

	td {
		text-align: right;
	}
```

cTopher commented on package.json:
```diff
@@ -0,0 +1,41 @@
+	},
+	"devDependencies": {
+		"@sveltejs/adapter-static": "^2.0.3",
+		"@sveltejs/kit": "^1.24.1",
```
This is quite an outdated version of svelte, if possible I would update to the latest version (you can check the zapper for a more up-to-date project)

cTopher commented on package.json:
```diff
@@ -0,0 +1,41 @@
+		"chart.js": "^4.4.3",
+		"chartjs-adapter-date-fns": "^3.0.0",
+		"date-fns": "^3.6.0",
+		"theoplayer": "^8.5.0"
```
This should be 8.8.3 (typings are broken in 8.5)

cTopher commented on package.json:
```diff
@@ -0,0 +1,41 @@
+		"preview": "vite preview",
+		"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
+		"check:watch": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --watch",
+		"lint": "prettier --check . && eslint .",
```
There are still some lint errors that should be fixed

cTopher reviewed (commented):
We also need a way of deploying this to deploy this. Based on https://bitbucket.org/r8szq0vy/theolive-player/src/master/web/demo/scripts/deploy.ts you could make a github action.

dubeyShivank commented:
> We also need a way of deploying this to deploy this. Based on https://bitbucket.org/r8szq0vy/theolive-player/src/master/web/demo/scripts/deploy.ts you could make a github action.

Do we want to deploy from here or from our internal repo? I removed the steps and the deploy scripts assuming we'd want to keep deployments from the internal repo but I'm ok to do it either way.

cTopher commented:
> Do we want to deploy from here or from our internal repo? I removed the steps and the deploy scripts assuming we'd want to keep deployments from the internal repo but I'm ok to do it either way.

I would do it here and fully remove all demo code from the old repo once this is merged. Make sure that all data that should not be publicly visible to be inside secrets.

cTopher commented:
![image](https://github.com/user-attachments/assets/4907485f-edd9-4fc2-b16f-7f398a6c529e)

For the chart I would:

-  increase the size on big screens 
-  hide the legend
-  reduce the amount of x axis labels
-  set min y value to 0 (there should not negative values)
-  remove opacity

cTopher commented on src/lib/Player.svelte:
```diff
@@ -0,0 +1,58 @@
+		if (onPlaying) {
+			newPlayer.addEventListener('playing', onPlaying)
+		}
+		player.source = {
```
This should happen on channel change (bug when navigating to the root url)

dubeyShivank commented:
> ![image](https://private-user-images.githubusercontent.com/4076701/401535766-4907485f-edd9-4fc2-b16f-7f398a6c529e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0Njg5MTgsIm5iZiI6MTczNjQ2ODYxOCwicGF0aCI6Ii80MDc2NzAxLzQwMTUzNTc2Ni00OTA3NDg1Zi1lZGQ5LTRmYzItYjE2Zi03ZjM5OGE2YzUyOWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTBUMDAyMzM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODkxMmRmNzRhMWI5YWQ4YTg2NTUxOTAwODQwMGFmMzFkMTJmYTM3YzM0MThlMWFiMzAyYjZmMWJjMTkwNWY3NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.pFWVddovNbzhUe4thaOpsVwUswEYAAYfZtEdLakNopc)
> 
> For the chart I would:
> 
> * increase the size on big screens
> * hide the legend
> * reduce the amount of x axis labels
> * set min y value to 0 (there should not negative values)
> * remove opacity

Is this better? 
![image](https://github.com/user-attachments/assets/577f90cf-2e7f-4ab1-addc-4935c735ed76)
![image](https://github.com/user-attachments/assets/1933cb74-8926-4c15-b730-b16c6a2a3529)

On a side note - on small screens if the demo has been running for a while, the graph is unreadable. I suggest we should have a max of the last 20s(?) worth of data points. What do you think?

dubeyShivank commented:
> > Do we want to deploy from here or from our internal repo? I removed the steps and the deploy scripts assuming we'd want to keep deployments from the internal repo but I'm ok to do it either way.
> 
> I would do it here and fully remove all demo code from the old repo once this is merged. Make sure that all data that should not be publicly visible to be inside secrets.

Easy, I've added a basic deployment script now, which will deploy on merge to `main`. But we can play around with the specific rules on when to deploy and what our approach to auto-updating version numbers is.

dubeyShivank commented on src/routes/+page.svelte:
```diff
@@ -0,0 +1,65 @@
+	onMount(() => {
+		if (!browser) return
+		const searchParams = new URLSearchParams(window.location.search)
+		channelId = searchParams.get('channel') ?? undefined
```
Should be okay now :)

cTopher commented on src/lib/Stats.svelte:
```diff
@@ -0,0 +1,95 @@
+		border-radius: 4px;
+		z-index: 1001;
+		left: 0 !important;
+		background-color: rgba(255, 255, 255, 0.3);
```
The text is still invisible on bright video's. RGB should be 0,0,0

cTopher commented on src/lib/Timeline.svelte:
```diff
@@ -0,0 +1,121 @@
+
+<style>
+	.container {
+		background-color: rgba(255, 255, 255, 0.5);
```
Same remark: still not visible on a bright video, RGB should be 0,0,0

cTopher reviewed (approved):
Feel free to merge!
